apiserver/facades/client/resources/facade.go

```diff
@@ -345,3 +278,25 @@ func errorResult(err error) params.ResourcesResult {
 		},
 	}
 }
+
+func convertParamsOrigin(origin params.CharmOrigin) corecharm.Origin {
+	var track string
+	if origin.Track != nil {
+		track = *origin.Track
+	}
+	return corecharm.Origin{
+		Source:   corecharm.Source(origin.Source),
```

Comment:
Missing type. Pretty sure it will be `charm`, but we should add it.

Reply:
done